src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -83,11 +83,33 @@ uct_rdmacm_cm_device_context_init(uct_rdmacm_cm_device_context_t *ctx,
     uint64_t general_obj_types_caps;
     ucs_status_t status;
     void *cap;
-    int ret;
 #endif
+    struct ibv_port_attr port_attr;
+    struct ibv_device_attr dev_attr;
+    int ret;
+    int i;
 
     ctx->num_dummy_qps = 0;
 
+    ret = ibv_query_device(verbs, &dev_attr);
+    if (ret != 0) {
+        ucs_error("ibv_query_device(%s) failed: %m", dev_name);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    ctx->is_eth = 0;
+    for (i = 0; i < dev_attr.phys_port_count; ++i) {
+        ret = ibv_query_port(verbs, i + 1, &port_attr);
+        if (ret != 0) {
+            ucs_error("ibv_query_port (%s) failed: %m", dev_name);
+            return UCS_ERR_IO_ERROR;
+        }
+
+        if (IBV_PORT_IS_LINK_LAYER_ETHERNET(&port_attr)) {
+            ctx->is_eth |= UCS_BIT(i);
```

Comment:
doing this in the loop seems weird to me, if different ports of the same device can be configured differently then this is a bug.
maybe initialize by first port and assert that others have same config?

Reply:
is_eth is bitset